### DISCUSSION TITLE
React-Query TypeScript Errors

### DISCUSSION THREAD

**Discussion by Asudharshanam** [TanStack/query#3073] [Q&A] [answered]
Hi!

Versions
````
"react-query": "3.34.0",
"axios": "0.22.0",
"typescript": "^3.8.3",
````

**Usage:**

index.js -- wrapper
```` 
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: 0,
      refetchInterval: 0,
      refetchOnWindowFocus: false
    }
  }
});

 <QueryClientProvider client={queryClient}>
      <App />
 </QueryClientProvider>
````

api:

````

export async function getTestData(
  id: number,
  abortSignal?: AbortSignal
) {
  const response = await apiInstance().get<TestData[]>(`${webApiUri}/test/${id}`,{ 
     signal: abortSignal 
   });
  return response.data;
}

export function useTestData(id: number) {
  return useQuery(
    ['test-data-by-id', id],
    ({ signal }) => getTestData(id, signal),
    {
      enabled: !!id,
      onError: () =>
        toastError('Error fetching data')
    }
  );
}
````

component:

````
  const { isLoading, data } = useTestData(id!);
````

**Everything works great**

BUT

````
"scripts": {
   "typecheck": "tsc --noEmit",
}
````

When I run `npm run typecheck,` I see a lot of errors with `node_modules/react-query/types/react/useQueries.d.ts`

![Screen Shot 2021-12-08 at 10 53 50 AM](https://user-images.githubusercontent.com/31739801/145250527-6321b7ba-f6c0-411a-b2d8-9f30a0c3c1c7.png)
![Screen Shot 2021-12-08 at 10 54 25 AM](https://user-images.githubusercontent.com/31739801/145250545-d82a7235-72c2-431d-8486-db0771da0c6a.png)
![Screen Shot 2021-12-08 at 10 54 56 AM](https://user-images.githubusercontent.com/31739801/145250564-08233653-1922-4b13-b601-aeb37dcfc1e4.png)
![Screen Shot 2021-12-08 at 10 55 12 AM](https://user-images.githubusercontent.com/31739801/145250575-1a524e96-1023-4998-a447-518aebf42ee4.png)

As a workaround added `skipLibCheck = true` in `tsconfig.json` for now.

Is there a way to fix this apart from `skipLibCheck = true`.

Also if there is anything wrong in terms of setup/coding standpoint or any version update needed. Please let me know.

Thank you 🙏

**Comment by TkDodo** [comment, marked as answer]
hm, really not sure what's going on here. We've explicitly added different types for ts < 4.1 when we made the change to useQueries so that smaller TS versions are not affected, as we use TS 4.1 features to achieve this:

https://github.com/tannerlinsley/react-query/blob/05a5382e4d73db46b4f562c30afb353fd0b165f8/package.json#L86-L92

could you provide a codesandbox reproduction?

also, just to let you know, we might drop this in the next major and bump the required TS version to 4.1 (which is over a year old as of now)

**Reply by Asudharshanam** [reply]
Unable to reproduce codesandbox. 

Will look forward for the `TS version 4.1`, that might solve the TS errors we are seeing. 

Thank you!

**Reply by Asudharshanam** [reply]
Upgraded TS Version and flxed the ts errors.
